pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1052,25 +1055,37 @@ func getTaskrunWorkspaces(ctx context.Context, pr *v1beta1.PipelineRun, rpt *res
 	return workspaces, pipelinePVCWorkspaceName, nil
 }
 
-// taskWorkspaceByWorkspaceVolumeSource is returning the WorkspaceBinding with the TaskRun specified name.
-// If the volume source is a volumeClaimTemplate, the template is applied and passed to TaskRun as a persistentVolumeClaim
-func taskWorkspaceByWorkspaceVolumeSource(wb v1beta1.WorkspaceBinding, taskWorkspaceName string, pipelineTaskSubPath string, owner metav1.OwnerReference) v1beta1.WorkspaceBinding {
+// taskWorkspaceByWorkspaceVolumeSource is returning the WorkspaceBinding for TaskRuns.
+// If the WorkspaceBinding a volumeClaimTemplate, an associated existing persistentVolumeClaim is binded to the TaskRun,
+// otherwise, the PipelineRun WorkspaceBinding is applied and binded to the TaskRun.
+func (c *Reconciler) taskWorkspaceByWorkspaceVolumeSource(ctx context.Context, pipelineWorkspaceName string, prName string, wb v1beta1.WorkspaceBinding, taskWorkspaceName string, pipelineTaskSubPath string, owner metav1.OwnerReference) (v1beta1.WorkspaceBinding, error) {
```

Comment:
Why does this function return an error now? It doesn't seem to be used.

Reply:
I probably did some experiment but forgot to put it back 😞 ... Fixed it.